### PR TITLE
DiscretizeOccupanciesTransformation: fix_denominator

### PR DESCRIPTION
**Pull request by josuav1**
added fix_denominator (bool) to DiscretizeOccupanciesTransformation: 

if True, will enforce a common denominator for all species. This prevents a mix of denominators (for example, 1/3, 1/4) that might require large cell sizes to perform an enumeration.

added Test; successful

### CODE REVIEW

File: pymatgen/transformations/standard_transformations.py [hunk 1]
```diff
@@ -696,13 +696,18 @@ def __init__(self, max_denominator=5, tol=0.25):
         self.max_denominator = max_denominator
         self.tol = tol
 
-    def apply_transformation(self, structure):
+    def apply_transformation(self, structure,fix_denominator=False):
```
Comment (shyuep) [inline pymatgen/transformations/standard_transformations.py]: This breaks the API for apply_transformation, and violates PEP8. If anything, this parameter should be part of the init.

File: pymatgen/transformations/standard_transformations.py [hunk 2]
```diff
@@ -716,7 +721,9 @@ def apply_transformation(self, structure):
                 old_occ = sp[k]
                 new_occ = float(
                     Fraction(old_occ).limit_denominator(self.max_denominator))
-                if round(abs(old_occ - new_occ), 6) > (
+                if fix_denominator: 
+                        new_occ = round(old_occ*self.max_denominator)/self.max_denominator
```
Comment (shyuep) [inline pymatgen/transformations/standard_transformations.py]: This is a bad implementation. What if the rounding is out of tolerance and fix_denominator is true? That will round the occupancies no matter what and by pass the check in the elif statement.

File: pymatgen/transformations/standard_transformations.py (hunk 1 above)
Comment (josuav1) [reply pymatgen/transformations/standard_transformations.py]: Thanks for the comment, fixed that.

File: pymatgen/transformations/standard_transformations.py (hunk 2 above)
Comment (josuav1) [reply pymatgen/transformations/standard_transformations.py]: It is unlikely someone wants to use a fixed_denominator only within a given tolerance, but I changed this now so there is no bypass of the tolerance anymore. Hope I fixed the bad implementation that way.

**Comment by shyuep** [comment]
Thanks for the fixes.

**Comment by shyuep** [comment]
There are failed tests on Py2 or 3. Pls fix.

https://circleci.com/gh/materialsproject/pymatgen/7263?utm_campaign=build-failed&utm_medium=email&utm_source=notification

**Pull request merged**